RubeesEat/Model/InMemoryBillRepository.cs

```diff
@@ -29,4 +29,32 @@ public void Add(Bill bill)
         ArgumentNullException.ThrowIfNull(bill);
         _bills.Add(bill.Id, bill);
     }
+
+    public IReadOnlyList<Bill> GetAllForUser(Guid userGuid)
+    {
+        return (from bill in _bills
+            where Enumerable.Any(bill.Value.EntryLines, line => line.Person.Id == userGuid)
+            select bill.Value).ToList();
```

Comment:
There two kind of writing LINQ queries. One is this query syntax you've used here. This is similar to SQL and allows stuff like filtering, selection, grouping, joining, etc.

The other way is using the method syntax. See this example on the differences in style: https://learn.microsoft.com/en-us/dotnet/csharp/linq/get-started/write-linq-queries#standard-query-operator-extension-methods

We don't really use the query syntax but instead use the method syntax. So your example would be:

```c#
return _bills.Values
    .Where(e => e.EntryLines.Any(f => f.Person == person)
    .ToList();
```

In this context I want you to take a look at two concepts that are relevant here:

 - The interface `IEnumerable<T>` and how to use it
 - Extension methods (https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods) and how they relate to LINQ